mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -10,63 +10,146 @@
 
 using namespace mlir;
 
-LogicalResult Conv2dGenerator::parseConvDims(
-    std::string &inputLayout, std::string &outputLayout,
-    std::string &filterLayout, int64_t batchSize, int64_t inputChannel,
-    int64_t inputHeight, int64_t inputWidth, int64_t outputChannel,
-    int64_t outputHeight, int64_t outputWidth, int64_t filterHeight,
-    int64_t filterWidth, SmallVector<int64_t, 4> &filterDimension,
-    SmallVector<int64_t, 4> &inputDimension,
-    SmallVector<int64_t, 4> &outputDimension) {
-  // Determine dimensions.
-  for (size_t i = 0; i < 4; ++i) {
-    auto &filterDim = filterLayout[i];
-    auto &inputDim = inputLayout[i];
-    auto &outputDim = outputLayout[i];
-
-    if (filterDim == 'k') {
-      filterDimension.push_back(outputChannel);
-    } else if (filterDim == 'c') {
-      filterDimension.push_back(inputChannel);
-    } else if (filterDim == 'y') {
-      filterDimension.push_back(filterHeight);
-    } else if (filterDim == 'x') {
-      filterDimension.push_back(filterWidth);
+void strToTokens(const std::string &arguments,
+                 std::map<std::string, std::string> &argMap) {
+  std::istringstream iss(arguments);
+  std::string token;
+  std::string argKey;
+  while (iss >> token) {
+    auto pos = token.find("--");
+    if (pos != std::string::npos) {
+      argKey = token.substr(pos + 2);
+    } else {
+      if (!argKey.empty()) {
+        argMap[argKey] = token;
+        argKey.clear();
+      }
     }
+  }
+}
```

Comment:
Since we do not plan to expose this in the header, make it wrapped in a anonymous namespace.